main.py

```diff
@@ -55,35 +56,52 @@ def cards(pages, columns=1, style="cards", force_version=False):
             pages = [pages]
         cards = []
         for page in pages:
-            with open("docs/%s.md" % page, "r") as doc_file:
-                doc = doc_file.read()
-                match = re.search("^# (.*)", doc, re.MULTILINE)
-                if match:
-                    header = match.groups()[0]
-                else:
-                    header = ""
-                default_meta = {
-                    "title": header,
-                    "short": "",
-                    "description": ""
-                }
-                doc_meta = {
-                    **default_meta,
-                    **meta.get_data(doc)[1]
-                }
-                cards.append(
-                    CARDS_TEMPLATE % (
-                        '/'.join((
-                            '',
-                            site,
-                            language,
-                            version,
-                            page
-                        )),
-                        doc_meta['short'] or doc_meta['title'],
-                        doc_meta['description'] or "&nbsp;"
-                        # site_url
-                    )
+            if page.startswith("https://") or page.startswith("http://"):
+                with urllib.request.urlopen(page) as file:
+                    content = file.read().decode('utf-8')
+                    match = re.search("<title>(.*)</title>", content, re.MULTILINE)
+                    if match:
+                        title = match.groups()[0]
+                    else:
+                        title = ""
+                    match = re.search("<meta name=\"description\" content=\"(.*)\"", content, re.MULTILINE)
```

Comment:
```suggestion
                    match = re.search("<meta property=\"og:title\" content=\"(.*)\"", content, re.MULTILINE)
                    if match:
                        title = match.groups()[0]
                    else:
                        title = ""
                    match = re.search("<meta property=\"og:description\" content=\"(.*)\"", content, re.MULTILINE)
```

Reply:
TODO: For security, make sure it's target is in a whitelist of hosts/domains (w/ only ibexa.co in it)

Reply:
As the white list is also public, we must be vigilant about main.py modification. Do not open a readthedocs preview from a non-Ibexa contributor modifying main.py